src/control/KinematicSimulationTrajectoryExecutor.cpp

```diff
@@ -63,7 +63,7 @@ void KinematicSimulationTrajectoryExecutor::validate(
 
 //==============================================================================
 std::future<void> KinematicSimulationTrajectoryExecutor::execute(
-    trajectory::TrajectoryPtr traj)
+    trajectory::ConstTrajectoryPtr traj)
```

Comment:
Since we don't allow `nullptr`, let's throw an exception for `nullptr`.

Please add this to other `[~]::execute()` functions.

Reply:
`execute` calls validate(), so the current implementation has execute take `const trajectory::ConstTrajectoryPtr` and pass the stored pointer to `validate` which checks for nullity. I think this considers both of your suggestions made on `validate` and `execute`.